### PR TITLE
Medical - Add setting to use limb damage for sum of trauma

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Title.
- Cleanup unused vars in handleIncapacitation.

Implementation is a bit lazy:
* You can't set limb DT per-unit (would be neat?)

Damage threshold is per-limb because configuring this as a user is a nightmare otherwise. Default is pretty generous to compensate for lack of armor in the limbs of most units.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/medical_damage/stringtable.xml [hunk 1]
```diff
@@ -843,5 +843,17 @@
             <Korean>치명상으로 인해 사망할 확률을 정합니다.</Korean>
             <Portuguese>A chance de morrer para um ferimento fatal.</Portuguese>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_uselimbDamage_DisplayName">
+            <English>Use Limb Damage</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_uselimbDamage_Description">
+            <English>Controls whether limb damage is taken into account for fatality and unconsciousness calculations.</English>
```
Comment (smitt14ua) [inline addons/medical_damage/stringtable.xml:850]:
```suggestion
            <English>Controls whether limb damage is taken into account for fatality and unconsciousness calculations.</English>
            <Russian>Определяет, учитывается ли повреждение конечностей при расчете смертности и потери сознания.</Russian>
```

File: addons/medical_damage/stringtable.xml [hunk 2]
```diff
@@ -843,5 +843,17 @@
...
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_limbDamageThreshold_DisplayName">
+            <English>Limb Damage Threshold</English>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_limbDamageThreshold_Description">
+            <English>Sets the amount of damage a limb can receive before going critical (leading to unconsciousness, or death if "Sum of Trauma" is enabled).\nDoes nothing if the "Use Limb Damage" setting is disabled.</English>
```
Comment (smitt14ua) [inline addons/medical_damage/stringtable.xml:856]:
```suggestion
            <English>Sets the amount of damage a limb can receive before going critical (leading to unconsciousness, or death if "Sum of Trauma" is enabled).\nDoes nothing if the "Use Limb Damage" setting is disabled.</English>
            <Russian>Устанавливает величину урона, который может получить конечность прежде, чем он станет критическим (приводит к потере сознания или смерти, если включена "Сумма травм").\nНичего не делает, если настройка "Использовать урон конечностям" отключена.</Russian>
```

**Comment by smitt14ua** [comment]
Looks like very useful PR for me. Sum of trauma is the most general reason of deaths and we have many complains about it

File: addons/medical_damage/initSettings.inc.sqf [hunk 3]
```diff
@@ -25,6 +25,24 @@
     true
 ] call CBA_fnc_addSetting;
 
+[
+    QEGVAR(medical,useLimbDamage),
+    "LIST",
+    [LSTRING(useLimbDamage_DisplayName), LSTRING(useLimbDamage_Description)],
+    ELSTRING(medical,Category),
+    [[0, 1, 2], [ELSTRING(common,Never), ELSTRING(common,aiOnly), ELSTRING(common,playersAndAI)], 0],
+    true
```
Comment (johnb432) [inline addons/medical_damage/initSettings.inc.sqf:34]:
```suggestion
    1
```

File: addons/medical_damage/initSettings.inc.sqf [hunk 4]
```diff
@@ -25,6 +25,24 @@
...
+] call CBA_fnc_addSetting;
+
+[
+    QEGVAR(medical,limbDamageThreshold),
+    "SLIDER",
+    [LSTRING(limbDamageThreshold_DisplayName), LSTRING(limbDamageThreshold_Description)],
+    ELSTRING(medical,Category),
+    [0, 25, 5, 2],
+    true
```
Comment (johnb432) [inline addons/medical_damage/initSettings.inc.sqf:43]:
```suggestion
    1
```

**Review by johnb432 — APPROVED** [review]
LGTM other than comments

File: addons/medical_damage/stringtable.xml [hunk 5]
```diff
@@ -843,5 +843,21 @@
             <Korean>치명상으로 인해 사망할 확률을 정합니다.</Korean>
             <Portuguese>A chance de morrer para um ferimento fatal.</Portuguese>
         </Key>
+        <Key ID="STR_ACE_Medical_Damage_uselimbDamage_DisplayName">
+            <English>Use Limb Damage</English>
+            <Russian>Использовать урон конечностям</Russian>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_uselimbDamage_Description">
+            <English>Controls whether limb damage is taken into account for sum of trauma calculations.</English>
+            <Russian>Определяет, учитывается ли повреждение конечностей при расчете смертности и потери сознания.</Russian>
```
Comment (smitt14ua) [inline addons/medical_damage/stringtable.xml:852]:
```suggestion
            <Russian>Определяет, учитывается ли повреждение конечностей при расчете совокупности травм.</Russian>
```

File: addons/medical_damage/stringtable.xml [hunk 6]
```diff
@@ -843,5 +843,21 @@
...
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_limbDamageThreshold_DisplayName">
+            <English>Limb Damage Threshold</English>
+            <Russian>Порог повреждения конечностей</Russian>
+        </Key>
+        <Key ID="STR_ACE_Medical_Damage_limbDamageThreshold_Description">
+            <English>Sets the amount of damage limbs can receive before going critical, leading to death. Must be over 0 for any effect.\nRequires the "Use Limb Damage" setting to be enabled and "Fatal Damage Source" to be set to "Sum of Trauma" or "Either".\nStacks multiplicatively with the overall unit Damage Threshold. Doesn't interact with fractures/limping at all.</English>
+            <Russian>Устанавливает величину урона, который может получить конечность, прежде чем он станет критическим (приводит к потере сознания или смерти, если включена "Сумма травм").\nНичего не делает, если настройка "Использовать урон конечностям" отключена.</Russian>
```
Comment (smitt14ua) [inline addons/medical_damage/stringtable.xml:860]:
```suggestion
            <Russian>Устанавливает величину урона, который могут получить конечности, прежде чем урон станет критическим, что приведет к смерти. Для любого эффекта должно быть больше 0.\nТребуется, чтобы настройка "Использовать урон конечностям" была включена, а "Причина смертельного урона" была установлена на "Совокупность травмы" или "Оба".\nУмножается с общим порогом повреждения. Никак не взаимодействует с переломами или хроманием.</Russian>
```

**Review by smitt14ua — CHANGES_REQUESTED** [review]
Updated Russian translations

File: addons/medical_gui/functions/fnc_updateBodyImage.sqf [hunk 7]
```diff
@@ -77,12 +77,22 @@ private _bodyPartBloodLoss = [0, 0, 0, 0, 0, 0];
         [_bloodLoss] call FUNC(bloodLossToRGBA);
     } else {
         private _damage = _bodyPartDamage select _forEachIndex;
+        // _damageThreshold here indicates how close unit is to guaranteed death via sum of trauma, so use the same multipliers used in medical_damage/functions/fnc_determineIfFatal.sqf
+        // TODO: make multipliers for head and torso a macro in medical_engine/script_macros_medical.hpp
         switch (true) do { // torso damage threshold doesn't need scaling
             case (_forEachIndex > 3): { // legs: index 4 & 5
-                _damageThreshold = LIMPING_DAMAGE_THRESHOLD * 4;
+                if (!EGVAR(medical,useLimbDamage) || EGVAR(medical,limbDamageThreshold) == 0) then { // Just indicate how close to the limping threshold we are
```
Comment (BrettMayson) [inline addons/medical_gui/functions/fnc_updateBodyImage.sqf:84]: This will throw an error if the setting is set to `2`

**Pull request merged**